examples/client/getModuleSource.ts

```diff
@@ -0,0 +1,62 @@
+import { ModuleReference } from '@concordium/common-sdk';
+import { createConcordiumClient } from '@concordium/node-sdk';
+import { credentials } from '@grpc/grpc-js';
+
+import meow from 'meow';
+
+const cli = meow(
+    `
+  Usage
+    $ yarn ts-node <path-to-this-file> [options]
+
+  Required:
+    --module, -m  The module reference of the module that you want the source for
+
+  Options
+    --help,     -h  Displays this message
+    --block,    -b  A block to query from, defaults to last final block
+    --endpoint, -e  Specify endpoint of the form "address:port", defaults to localhost:20000
+`,
+    {
+        importMeta: import.meta,
+        flags: {
+            module: {
+                type: 'string',
+                alias: 'm',
+                isRequired: true,
+            },
+            endpoint: {
+                type: 'string',
+                alias: 'e',
+                default: 'localhost:20000',
+            },
+            block: {
+                type: 'string',
+                alias: 'b',
+                default: '', // This defaults to LastFinal
+            },
+        },
+    }
+);
+
+const [address, port] = cli.flags.endpoint.split(':');
+const client = createConcordiumClient(
+    address,
+    Number(port),
+    credentials.createInsecure(),
+    { timeout: 15000 }
+);
+
+if (cli.flags.h) {
+    cli.showHelp();
+}
+
+/// Gets the source of a module on the chain.
+/// Note that this returns the raw bytes of the source, as a HexString.
+
+(async () => {
+    const ref = new ModuleReference(cli.flags.module);
+    const source = await client.getModuleSource(ref, cli.flags.block);
+
+    console.log(source);
```

Comment:
This just prints the beginning of the source:
```
<Buffer 00 61 73 6d 01 00 00 00 01 a8 01 19 60 02 7f 7f 00 60 04 7f 7f 7f 7f 00 60 02 7f 7f 01 7f 60 03 7f 7f 7f 01 7f 60 01 7e 01 7f 60 03 7f 7f 7f 01 7e 60 ... >
```

Maybe we can display it better. (or maybe we don't want to print the entire blob? :thinking: )

Reply:
Wait why does this even return a buffer when everything else returns a `HexString`? We should definitely display it better though, I'll make it print the blob as a hex string.

Reply:
I made it write to a file instead

Reply:
Should we? :thinking:  :shrug:

Reply:
Seems less simple to write to a file, but getting a very long hex/base64 output also seems wrong. To me it seemed least bad, but I really don't have strong opinions on it.

Reply:
Ah makes sense, I mean it's not ideal that it's a buffer when everything else is hex strings, right? On the other hand it's also bad to break compatibility.

It doesn't matter much though.